src/tests/suite/daos_degrade_ec.c

```diff
@@ -479,6 +478,72 @@ degrade_ec_partial_update_agg(void **state)
 	free(verify_data);
 }
 
+static void
+degrade_ec_agg(void **state)
+{
+	test_arg_t	*arg = *state;
+	struct ioreq	req;
+	daos_obj_id_t	oid;
+	d_rank_t	rank;
+	int		i;
+	char		*data;
+	char		*verify_data;
+
+	if (!test_runable(arg, 6))
+		return;
+
+	data = (char *)malloc(EC_CELL_SIZE);
+	assert_true(data != NULL);
+	verify_data = (char *)malloc(EC_CELL_SIZE);
+	assert_true(verify_data != NULL);
+	oid = daos_test_oid_gen(arg->coh, OC_EC_4P2G1, 0, 0, arg->myrank);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+	for (i = 0; i < 4; i++) {
+		daos_recx_t recx;
+
+		req.iod_type = DAOS_IOD_ARRAY;
+		recx.rx_nr = EC_CELL_SIZE;
+		recx.rx_idx = i * EC_CELL_SIZE;
+		memset(data, 'a' + i, EC_CELL_SIZE);
+		insert_recxs("d_key", "a_key", 1, DAOS_TX_NONE, &recx, 1,
+			     data, EC_CELL_SIZE, &req);
+	}
+
+	for (i = 7; i >= 4; i--) {
+		daos_recx_t recx;
+
+		req.iod_type = DAOS_IOD_ARRAY;
+		recx.rx_nr = EC_CELL_SIZE;
+		recx.rx_idx = i * EC_CELL_SIZE;
+		memset(data, 'a' + i, EC_CELL_SIZE);
+		insert_recxs("d_key", "a_key", 1, DAOS_TX_NONE, &recx, 1,
+			     data, EC_CELL_SIZE, &req);
+	}
+
+	/* Trigger aggregation */
+	daos_pool_set_prop(arg->pool.pool_uuid, "reclaim", "time");
+	trigger_and_wait_ec_aggreation(arg, &oid, 1, false);
+
+	/* Kill one data shard to make sure the data is correct after
+	 * aggregation .
+	 */
+	rank = get_rank_by_oid_shard(arg, oid, 2);
+	rebuild_pools_ranks(&arg, 1, &rank, 1, false);
+	print_message("sleep 30 seconds for VOS aggregation on each xstream");
+	sleep(30);
+	trigger_and_wait_ec_aggreation(arg, &oid, 1, false);
```

Comment:
seems better to first do "trigger_and_wait_ec_aggreation", and then do sleep() to wait vos agg?
BTW, vos agg may need wait more than 40S to make sure it happens.

Reply:
Or better yet, come up with a way to check status so we don't sleep